modules/service/variables.tf

```diff
@@ -277,7 +277,7 @@ variable "container_definition_defaults" {
 variable "cpu" {
   description = "Number of cpu units used by the task. If the `requires_compatibilities` is `FARGATE` this field is required"
   type        = number
-  default     = 1024
```

Comment:
this would break the module as its currently written since its designed to be Fargate-first. However, you can set this to `null` in your configuration to disable the value when using EC2

Reply:
As a side note, there is also [`nullable` argument](https://developer.hashicorp.com/terraform/language/values/variables#disallowing-null-input-values) which may be helpful for some cases in the future (when we consider upgrading requirements for Terraform 1.1+ or further). Just saying :)